iceoryx_utils/include/iceoryx_utils/cxx/type_traits.hpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_UTILS_CXX_TYPE_TRAITS_HPP
+#define IOX_UTILS_CXX_TYPE_TRAITS_HPP
+
+#include <type_traits>
+
+namespace iox
+{
+namespace cxx
+{
+///
+/// @brief Verifies whether the passed Callable type is in fact callable
+///
+template <typename Callable, typename... ArgTypes>
+struct is_callable
```

Comment:
If implemented something similar inside the class `cxx::function_ref`. Could we have a look at it together and maybe combine it?